pkg/auth/rhsso_authz_handler.go

```diff
@@ -178,14 +196,18 @@ func (a *AuthzHandler) checkInfraEnvBasedAccess(id string, request *http.Request
 	}
 
 	if infraEnv.ClusterID != "" {
-		if isAllowed, _ := a.checkClusterBasedAccess(infraEnv.ClusterID.String(), request, payload); isAllowed {
+		if isAllowed, _ := a.checkClusterBasedAccess(infraEnv.ClusterID.String(), action, payload); isAllowed {
```

Comment:
We're invoking isObjectOwnedByUser here (as part of hasSubscriptionAccess), so in case isAllowed is false - we'll call isObjectOwnedByUser again in line 209. So to avoid that we can maybe do the 'action == ReadAction' first.
i.e. move lines 206-208 to 197. Then we won't need a fallback to checkClusterBasedAccess - just return isAllowed

Reply:
fixed